libcaf_net/test/datagram_transport.cpp

```diff
@@ -221,7 +222,7 @@ CAF_TEST(resolve and proxy communication) {
   auto mgr_impl = mgr.downcast<endpoint_manager_impl<transport_type>>();
   CAF_CHECK(mgr_impl != nullptr);
   auto& transport = mgr_impl->transport();
-  transport.add_new_worker(make_node_id(uri), ep);
+  CAF_CHECK(!transport.add_new_worker(make_node_id(uri), ep));
```

Comment:
For checking error codes, I much prefer comparing to `none`. Otherwise, it looks like the function returns a boolean.

```suggestion
  CAF_CHECK_EQUAL(transport.add_new_worker(make_node_id(uri), ep), none);
```